cloud/disk_manager/internal/pkg/services/snapshots/create_snapshot_from_disk_task.go

```diff
@@ -372,3 +377,36 @@ func (t *createSnapshotFromDiskTask) GetResponse() proto.Message {
 		StorageSize: t.state.SnapshotStorageSize,
 	}
 }
+
+////////////////////////////////////////////////////////////////////////////////
+
+func (t *createSnapshotFromDiskTask) validateCheckpointStatus(
+	ctx context.Context,
+	nbsClient nbs.Client,
+	diskID string,
+	checkpointID string,
+) error {
+
+	status, err := nbsClient.GetCheckpointStatus(ctx, diskID, checkpointID)
+	if err != nil {
+		return err
+	}
+
+	switch status {
+	case nbs.CheckpointStatusNotReady:
+		return errors.NewInterruptExecutionError()
+
+	case nbs.CheckpointStatusError:
+		err := nbsClient.DeleteCheckpoint(
+			ctx,
+			diskID,
+			checkpointID,
+		)
+		if err != nil {
+			return err
+		}
+		return errors.NewRetriableErrorf("Failed to create checkpoint for disk %s.", diskID)
```

Comment:
Do we need to wrap DeleteCheckpoint error?

Reply:
Decided not to check err from DeleteCheckpoint at all, just return RetriableError